### PR TITLE
 MM-1705 add google as an oauth single-sign-on service

### PR DESCRIPTION
**Pull request by jwilander**
This PR implements Google as an SSO service much like we did for GitLab before. Due to Google's strict OAuth APIs some small refactoring (mainly incorporating team names into the state field) was done to accommodate that.

There is also some reformatting to match the style guide and I re-added getMe as a synchronous call because that was mistakenly removed and caused some issues with OAuth login.

I edited the config for this.

### CODE REVIEW

**Comment by crspeller** [comment]
Looks good to me. Need a seconder.

File: web/react/utils/client.jsx [hunk 1]
```diff
@@ -279,24 +279,33 @@ module.exports.getAudits = function(userId, success, error) {
     });
 };
 
-module.exports.getMe = function(success, error) {
+module.exports.getMeSynchronous = function(success, error) {
+    var currentUser = null;
 
     $.ajax({
+        async: false,
```
Comment (rgarmsen2295) [inline web/react/utils/client.jsx]: This probably isn't the PR to change this in but just want to point out from #370 that the `async: false` setting is deprecated by jquery so we should probably find another way to do this if possible.

**Comment by rgarmsen2295** [comment]
Assuming we're okay with leaving the async option in for now, +1 from me.

File: web/react/utils/client.jsx (hunk 1 above)
Comment (jwilander) [inline web/react/utils/client.jsx]: It is not being deprecated. The use of ```aysnc: false``` with promises (.done(), .fail(), etc.) is being deprecated but we don't use promises here so it's fine. This is why I'm adding the code back in.

http://api.jquery.com/jquery.ajax/

Comment (rgarmsen2295) [inline web/react/utils/client.jsx]: Ah okay cool had no idea, reading #370 made it seem like it was all uses of it and figured someone originally removed it here for the same reason.

**Pull request merged**